pkg/kubelet/dockertools/convert.go

```diff
@@ -74,3 +80,31 @@ func toRuntimeImage(image *docker.APIImages) (*kubecontainer.Image, error) {
 		Size: image.VirtualSize,
 	}, nil
 }
+
+// convert ContainerStatus to api.ContainerStatus.
+func containerStatusToAPIContainerStatus(containerStatus *kubecontainer.ContainerStatus) *api.ContainerStatus {
```

Comment:
@Random-Liu OK I see. I think rkt needs the similar logic, and use a reason cache to change the status from `Terminted` to `Waiting`.
BTW why there is two `else if` [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockertools/manager.go#L553-L562)? Can we make it a function to check the reason ?